xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -2161,7 +2162,22 @@ public boolean equals(java.lang.Object arg0)
         Document d = (Document) arg0;
         return d.getXWikiContext().equals(getXWikiContext()) && this.doc.equals(d.doc);
     }
-
+    
+    @Override 
+    public int hashCode() 
+    {
+        return new HashCodeBuilder(3,17)
+            .append(this.initialDoc)
+            .append(this.doc)
+            .append(this.currentObj)
+            .append(this.currentMixedDocumentReferenceResolver)
+            .append(this.defaultEntityReferenceSerializer)
+            .append(this.localEntityReferenceSerializer)
+            .append(this.compactWikiEntityReferenceSerializer)
+            .append(this.documentRevisionProvider)
+            .append(this.authorizationManager)
+            .toHashCode();
```

Comment:
Adding some hashcode is always tricky so we would also need you to add some tests to prove that it works. Now I can see for sure that it's not correct :) since for example injected components have no reason to be class model data (they're are simply services used by the class).